core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,497 @@
+
+  async ngOnInit(): Promise<void> {
+    const userInfo = await this.userService.getUserInfoAsync();
+    const username = userInfo.getUsername();
+
+    if (username === null) {
+      throw new Error('Cannot fetch username.');
+    }
+    this.username = username;
+    const currentOption = this.options.filter(
+      (option) => option.contributionType === this.type);
+    this.selectedContributionType = currentOption[0].displayName;
+
+    const userContributionRights =
+      await this.userService.getUserContributionRightsDataAsync();
+    this.userCanReviewTranslationSuggestions = (
+      userContributionRights
+        .can_review_translation_for_language_codes.length > 0);
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    if (this.userCanReviewTranslationSuggestions) {
+      this.options.push(this.translationReviewOption);
+    }
+    if (this.userCanSuggestQuestions) {
+      this.options.push(this.questionContributionOption);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.options.push(this.questionReviewOption);
+    }
+
+    await this.fetchStats();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  toggleMobileDropdown(): void {
+    this.mobileDropdownShown = !this.mobileDropdownShown;
+  }
+
+  async selectOption(contributionType: string): Promise<void> {
+    this.type = contributionType;
+    const currentOption = this.options.filter(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption[0].displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    this.translationContributions = [];
+    this.translationReviews = [];
+    this.questionContributions = [];
+    this.questionReviews = [];
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      const translationContributionLanguages = this.fetchLanguagesFromStats(
+        response.translation_contribution_stats);
+      translationContributionLanguages.forEach((language: string) => {
+        const translationContributionsStats:
+        TranslationContributionStats[] = [];
+        response.translation_contribution_stats.map((stat) => {
+          if (stat.language_code === language) {
+            translationContributionsStats.push(
+              this.createTranslationContributionStat(stat)
+            );
+          }
+        });
+
+        const translationContributionSlicedStats: Stats[] =
+          this.sliceStatsIntoPage(translationContributionsStats);
+
+        this.translationContributionPagedStats.push(
+          this.generatePagedStats(
+            translationContributionSlicedStats, language));
```

Comment:
Would it be simpler to just have an array of stats objects and incrementing/decrementing the index by page size on next/previous page instead of using the `PagedStats` objects? That's what we do for the opportunity pagination currently.

Reply:
Explained above. I have modified the pagination logic.